decoder/block_candidates.go

```diff
@@ -54,18 +54,48 @@ func detailForBlock(block *schema.BlockSchema) string {
 	return strings.TrimSpace(detail)
 }
 
-func snippetForBlock(blockType string, block *schema.BlockSchema) string {
-	labels := ""
-	placeholder := 1
+func snippetForBlock(blockType string, block *schema.BlockSchema, prefillRequiredAttributes bool) string {
+	if prefillRequiredAttributes {
+		labels := ""
 
-	for _, l := range block.Labels {
-		if l.IsDepKey {
-			labels += fmt.Sprintf(` "${%d}"`, placeholder)
-		} else {
+		depKey := false
+		for _, l := range block.Labels {
+			if l.IsDepKey {
+				depKey = true
+			}
+		}
+
+		if depKey {
+			for _, l := range block.Labels {
+				if l.IsDepKey {
+					labels += ` "${0}"`
+				} else {
+					labels += fmt.Sprintf(` "%s"`, l.Name)
+				}
+			}
+			return fmt.Sprintf("%s%s {\n \n}", blockType, labels)
+		}
+
+		placeholder := 1
+		for _, l := range block.Labels {
 			labels += fmt.Sprintf(` "${%d:%s}"`, placeholder, l.Name)
+			placeholder++
 		}
-		placeholder++
-	}
 
-	return fmt.Sprintf("%s%s {\n  ${%d}\n}", blockType, labels, placeholder)
+		return fmt.Sprintf("%s%s {\n  ${%d}\n}", blockType, labels, placeholder)
+	} else {
+		labels := ""
+		placeholder := 1
+
+		for _, l := range block.Labels {
+			if l.IsDepKey {
+				labels += fmt.Sprintf(` "${%d}"`, placeholder)
+			} else {
+				labels += fmt.Sprintf(` "${%d:%s}"`, placeholder, l.Name)
+			}
+			placeholder++
+		}
+
+		return fmt.Sprintf("%s%s {\n  ${%d}\n}", blockType, labels, placeholder)
+	}
```

Comment:
```suggestion
	}
	
	labels := ""
	placeholder := 1

	for _, l := range block.Labels {
		if l.IsDepKey {
			labels += fmt.Sprintf(` "${%d}"`, placeholder)
		} else {
			labels += fmt.Sprintf(` "${%d:%s}"`, placeholder, l.Name)
		}
		placeholder++
	}

	return fmt.Sprintf("%s%s {\n  ${%d}\n}", blockType, labels, placeholder)
```
Nitpick: I think the else block here is redundant since you return early from the other branch.